src/App.tsx

```diff
@@ -1,25 +1,48 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import './App.css';
-import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const items = 42;
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialPage = Number(searchParams.get('page')) || 1;
+  const initialPerPage = Number(searchParams.get('perPage')) || 5;
+
+  const [perPage, setPerPage] = useState(initialPerPage);
+  const [currentPage, setCurrentPage] = useState(initialPage);
+
+  const start = (currentPage - 1) * perPage + 1;
+  const end = Math.min(currentPage * perPage, items);
+
+  useEffect(() => {
+    setSearchParams({
+      page: currentPage.toString(),
+      perPage: perPage.toString(),
+    });
+  }, [currentPage, perPage, setSearchParams]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${start} - ${end} of ${items})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            defaultValue="5"
```

Comment:
The `defaultValue` of the select element is set to '5', which is correct if the initial state for `perPage` is also 5. However, ensure that this matches the `initialPerPage` value derived from the URL parameters to avoid inconsistencies. Consider setting the `defaultValue` dynamically based on `initialPerPage`.